src/components/MainSidebar/DimensionsList/DimensionsList.js

```diff
@@ -0,0 +1,111 @@
+import i18n from '@dhis2/d2-i18n'
+import { CircularLoader, NoticeBox } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useRef, useEffect } from 'react'
+import { DimensionListItem } from './DimensionListItem.js'
+import styles from './DimensionsList.module.css'
+
+const getNoResultsMessage = (searchTerm, programName) => {
+    if (searchTerm) {
+        return i18n.t("No dimensions found for '{{searchTerm}}'", {
+            searchTerm,
+        })
+    }
+    if (programName) {
+        return i18n.t('No dimensions found in the {{programName}} program', {
+            programName,
+        })
+    }
+    return i18n.t('No dimensions found')
+}
+
+const DimensionsList = ({
+    loading,
+    fetching,
+    error,
+    dimensions,
+    programName,
+    searchTerm,
+    setIsListEndVisible,
+}) => {
+    const scrollBoxRef = useRef()
+    const updateIsListEndVisible = () => {
```

Comment:
the name `updateIsListEndVisible` sort of clashes with `setIsListEndVisible` (but looking at it further I understand that the former is a wrapper for the latter. Was thinking it might be called `onScroll`, but that might not fit so well when called from the `useEffect` below, or does it? 🤔

Reply:
I see what you mean. Tried to avoid naming collisions in 6ccaedc but not sure that's much better....

Reply:
haha, try reading `setIsListEndVisible(isScrollboxEndVisible(scrollBoxRef.current))` quickly 5 times in a row 😉  Hard to say, I think this way at least you can't mix up the two fns, but neither is a clear choice maybe. Up to you 👍